scripts/kill-host-pods.py

```diff
@@ -7,45 +7,69 @@
 import click
 
 CTR = os.path.expandvars("$SNAP/microk8s-ctr.wrapper")
-KUBECTL = [
-    "{}/kubectl".format(os.getenv("SNAP")),
-    "--kubeconfig={}/credentials/kubelet.config".format(os.getenv("SNAP_DATA")),
-]
+SNAP = os.getenv("SNAP")
+SNAP_DATA = os.getenv("SNAP_DATA")
+SNAP_DATA_CURRENT = os.path.abspath(f"{SNAP_DATA}/../current")
+KUBECTL = [f"{SNAP}/kubectl", f"--kubeconfig={SNAP_DATA}/credentials/kubelet.config"]
+
+
+def post_filter_has_known_containers(pod, containers: list) -> bool:
+    """
+    Return true if any of the container IDs on the pod match the list of
+    containers passed on the second argument.
+    """
+    for container in pod["status"]["containerStatuses"] or []:
+        try:
+            _, container_id = container["containerID"].split("containerd://", 2)
+            if container_id in containers:
+                return True
+        except (KeyError, ValueError, TypeError, AttributeError):
+            continue
+
+    return False
+
+
+def post_filter_has_snap_data_mounts(pod) -> bool:
+    """
+    Return true if a pod definition contains any volumes that mount paths
+    from `/var/snap/microk8s/current/...`, e.g. CNI pods
+    """
+    for volume in pod["spec"].get("volumes", []):
+        hostpath_volume = volume.get("hostPath", {})
+        if hostpath_volume.get("path", "").startswith(SNAP_DATA_CURRENT):
```

Comment:
Given that $SNAP_DATA would also be eventually removed, shouldn't we also check for `/var/snap/microk8s/any-revision/` ?

Reply:
I'd say we only really care about `/var/snap/microk8s/current`. `/var/snap/microk8s/common` paths for example do not have the same issue.

I don't we mount pods with specific `/var/snap/microk8s/$revision` anywhere, why do you think we need this?

Reply:
In the past we used to put into the manifests "$SNAP_DATA". I do not see this anymore, we use the path with "current". Do you think it would be much more work to catch those cases too? At least as a fail safe?

Reply:
i'd much rather we leave them as-is. especially in a cluster setting, this would not be a good idea.